Ray.jl/src/object_store.jl

```diff
@@ -17,12 +17,17 @@ end
 """
     Ray.get(object_id::ObjectIDAllocated)
 
-Retrieves the data associated with the `object_id` from the object store.
-This method is blocking until the data is available in the local object store.
+Retrieves the data associated with the `object_id` from the object store.  This
+method is blocking until the data is available in the local object store, even
+if run in an `@async` task.
+
+If the task that generated the `ObjectID` failed with a Julia exception, the
+captured exception will be thrown on `get`.
 """
 function get(oid::rayjll.ObjectIDAllocated)
     io = IOBuffer(take!(rayjll.get(oid)))
-    return deserialize(io)
+    result = deserialize(io)
+    result isa RayRemoteException ? throw(result) : return result
```

Comment:
We'll probably want to be able to allow the caller to have the exception just be returned if they wish. I do think that this should be the default though and we can add that functionality in later

Reply:
I'll add a TODO